File: Zaoch_2011/D.py
import fractions
from decimal import *

l = int(Decimal(input()) * 1000000000)
a = int(Decimal(input()) * 1000000000)
b = int(Decimal(input()) * 1000000000)
lcd = a * b // fractions.gcd(a, b)
print(l // lcd - (1 if l % lcd == 0 else 0))

File: Zaoch_2011/D.java
import java.math.*;
import java.util.Scanner;

public class D {
	public static void main(String[] args) {
		Scanner in = new Scanner(System.in);
		BigInteger l = readNumber(in), a = readNumber(in), b = readNumber(in);
		BigInteger lcd = (a.multiply(b)).divide(a.gcd(b));
		System.out.println(l.divide(lcd).subtract(l.mod(lcd).equals(BigInteger.ZERO) ? BigInteger.ONE : BigInteger.ZERO));
	}

	static BigInteger readNumber(Scanner in) {
		String s = in.nextLine();
		if (s.charAt(0) == '.')
			s = "0" + s;
		return new BigDecimal(s).multiply(BigDecimal.valueOf(1000000000)).toBigInteger();
	}
}